schema/status_schema_0-0-0.json

```diff
@@ -0,0 +1,241 @@
+          }
+        },
+        "memory": {
+          "description": "RAM usage, in bytes, as a 2-tuple: [total_ram, available_ram]",
+          "type": "array",
+          "items": { "type": "integer" },
+          "minItems": 2,
+          "maxItems": 2
+        },
+        "storage": {
+          "description": "writable storage for data, as a 2-tuple: [total_space, free_space]",
+          "type": "array",
+          "items": { "type": "integer" },
+          "minItems": 2,
+          "maxItems": 2
+        },
+        "load": {
+          "description": "CPU load, as a 3-tuple: [1min_load, 5min_load, 15min_load]",
+          "type": "array",
+          "items": {
+            "type": "number",
+            "minimum": 0.0
+          },
+          "minItems": 3,
+          "maxItems": 3
+        }
+      }
+    },
+    "timeSources": {
+      "description": "status of time-keeping services",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "oneOf": [
+          { "$ref": "#/$defs/ntpTimeSource" },
+          { "$ref": "#/$defs/ppsTimeSource" },
+          { "$ref": "#/$defs/nmeaTimeSource" }
+        ]
+      }
+    },
+    "cellular": {
+      "description": "cellular connections",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "description": "cellular connection status",
+        "required": [
+          "hardware"
+        ],
+        "properties": {
+          "hardware": {
+            "description": "harddware informatsion",
+            "type": "object",
+            "properties": {
+              "vendor": { "type": "string" },
+              "model": { "type": "string" },
+              "revision": { "type": "string" }
+            }
+          },
+          "operator": {
+            "description": "cellular operator",
+            "type": "object",
+            "properties": {
+              "code": { "type": "string" },
+              "name": { "type": "string" }
+            }
+          },
+          "connection": {
+            "description": "connection details",
+            "type": "object",
+            "properties": {
+              "technology": {
+                "type": "string",
+                "examples": [
+                  "gsm", "cdma1x", "evdo", "umts", "lte", "5G"
+                ]
+              },
+              "signal": {
+                "description": "signal quality metrics",
+                "type": "object",
+                "properties": {
+                  "ecio": { "type": "number" },
+                  "io": { "type": "number" },
+                  "rscp": { "type": "number" },
+                  "rsrp": { "type": "number" },
+                  "rsrq": { "type": "number" },
+                  "rssi": { "type": "number" },
+                  "sinr": { "type": "number" },
+                  "snr": { "type": "number" }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "$defs": {
```

Comment:
I'd put use snake_case everywhere to match the rest of our messages (even though I prefer camelCase...)

Reply:
I initially used camelCase because the schema itself uses camelCase for the keywords: `oneOf`, `allOf` and so on, so I wanted to keep this consistent.

But consistency with our other messages is indeed stronger, so I'll switch to camelCase, except in places where it does not make sense.

Reply:
Oh, by the way: everything that is below `$defs` is a definition, like a type definition,
like we extend the basic known types (`boolean`, `integer`, `number`, `string`, `array`, `object`...).

So, `ntpTimeSource` fo rexample, is not a field of the message, but a definition to be
re-used elsewhere. And we use it to define the `time_source` field; that is, we use it
as the type for the field.

Do you think snake_case is still mandatory even for `$defs`, or do you think camelCase
is acceptable? I would prefer to use camelCase in that situation, nd keep snake_case for
the actual fields of messages.